guides/theming.md

```diff
@@ -1,311 +1,359 @@
+_multiple_ themes by repeating the API calls multiple times. You can do this either in the same
+theme file or in separate theme files.
+
+#### Multiple themes in one file
 
-### Theming only certain components
-
-The `angular-material-theme` mixin will output styles for [all components in the library](https://github.com/angular/components/blob/master/src/material/core/theming/_all-theme.scss).
-If you are only using a subset of the components (or if you want to change the theme for specific
-components), you can include component-specific theme mixins. You also will need to include
-the `mat-core-theme` mixin as well, which contains theme-specific styles for common behaviors
-(such as ripples).
-
- ```scss
-@import '~@angular/material/theming';
-// Plus imports for other components in your app.
-
-// Include the common styles for Angular Material. We include this here so that you only
-// have to load a single css file for Angular Material in your app.
-// **Be sure that you only ever include this mixin once!**
-@include mat-core();
-
-// Define the theme.
-$candy-app-primary: mat-palette($mat-indigo);
-$candy-app-accent:  mat-palette($mat-pink, A200, A100, A400);
-$candy-app-theme:   mat-light-theme((
-  color: (
-    primary: $candy-app-primary,
-    accent: $candy-app-accent,
-  ),
-  typography: mat-typography-config()
+Defining multiple themes in a single file allows you to support mutliple themes without having to
+manage loading of multiple CSS assets. The downside, however, is that your CSS will include more
+styles than necessary.
+
+To control which theme applies when, `@include` the mixins only within a context specified via
+CSS rule declaration. See the [documentation for Sass mixins][sass-mixins] for further background.
+
+[sass-mixins]: https://sass-lang.com/documentation/at-rules/mixin
+
+```scss
+@use '~@angular/material' as mat;
+
+// Define a light theme
+$light-primary: mat.define-palette(mat.$indigo-palette);
+$light-accent: mat.define-palette(mat.$pink-palette);
+$light-theme: mat.define-light-theme((
+ color: (
+   primary: $light-primary,
+   accent: $light-accent,
+ )
 ));
 
-// Include the theme styles for only specified components.
-@include mat-core-theme($candy-app-theme);
-@include mat-button-theme($candy-app-theme);
-@include mat-checkbox-theme($candy-app-theme);
+// Define a dark theme
+$dark-primary: mat.define-palette(mat.$pink-palette);
+$dark-accent: mat.define-palette(mat.$bluegray-palette);
+$dark-theme: mat.define-dark-theme((
+ color: (
+   primary: $dark-primary,
+   accent: $dark-accent,
+ )
+));
+
+// Apply the dark theme by default
+@include mat.core-theme($dark-theme);
+@include mat.button-theme($dark-theme);
+
+// Apply the light theme only when the `.my-light-theme` CSS class is applied
+// to an ancestor element of the components (such as `body`).
+.my-light-theme {
+ @include mat.core-theme($light-theme);
+ @include mat.button-theme($light-theme);
```

Comment:
Could this be `button-color` to reduce unnecessary typography duplication?

Reply:
Yes, I actually meant to use `-color` here. However, in this example the typography styles wouldn't be emitted anyway since there's no typography config in the theme definition.